src/opensles/AudioInputStreamOpenSLES.cpp

```diff
@@ -124,11 +126,29 @@ Result AudioInputStreamOpenSLES::open() {
     SLresult result = EngineOpenSLES::getInstance().createAudioRecorder(&mObjectInterface,
                                                                         &audioSrc,
                                                                         &audioSink);
+
+    // If the result is "invalid parameter" and we're using the extended data format it could be
+    // that the underlying hardware doesn't support it
+    // (e.g. Moto G: https://github.com/google/oboe/issues/192)
+    // Try again using the old format
+    if (result == SL_RESULT_PARAMETER_INVALID && usingExtendedDataFormat){
+
+        LOGW("Unable to open recording stream using PCM_EX format, trying again using PCM...");
+        audioSink.pFormat = &format_pcm;
```

Comment:
Should we set usingExtendedDataFormat=false here? For line 150 below.

Reply:
Doh! Thanks Phil - this is exactly why I ask you to review :)

Reply:
Fixed now